pkg/api/health.go

```diff
@@ -129,8 +139,184 @@ func PrintOverallReleaseHealth(w http.ResponseWriter, curr, twoDay, prev sippypr
+	infraIndicator, err := getIndicatorForTest(dbc, release, infraTestName)
+	if err != nil {
+		klog.Errorf("error querying test report: %s", err)
+		return
+	}
+	indicators["infrastructure"] = infraIndicator
+
+	// Install
+	installTestName := strings.TrimPrefix(testgridanalysisapi.InstallTestName,
+		testgridanalysisapi.SippySuiteName+".")
+	installIndicator, err := getIndicatorForTest(dbc, release, installTestName)
+	if err != nil {
+		klog.Errorf("error querying test report: %s", err)
+		return
+	}
+	indicators["install"] = installIndicator
+
+	// Upgrade
+	upgradeTestName := strings.TrimPrefix(testgridanalysisapi.UpgradeTestName,
+		testgridanalysisapi.SippySuiteName+".")
+	upgradeIndicator, err := getIndicatorForTest(dbc, release, upgradeTestName)
+	if err != nil {
+		klog.Errorf("error querying test report: %s", err)
+		return
+	}
+	indicators["upgrade"] = upgradeIndicator
+
+	// Tests
+	// NOTE: this is not actually representing the percentage of tests that passed, it's representing
+	// the percentage of time that all tests passed. We should probably fix that.
+	testsTestName := strings.TrimPrefix(testgridanalysisapi.OpenShiftTestsName,
+		testgridanalysisapi.SippySuiteName+".")
+	testsIndicator, err := getIndicatorForTest(dbc, release, testsTestName)
+	if err != nil {
+		klog.Errorf("error querying test report: %s", err)
+		return
+	}
+	indicators["tests"] = testsIndicator
+
+	var lastUpdated time.Time
+	r := dbc.DB.Raw("SELECT MAX(created_at) FROM prow_job_runs").Scan(&lastUpdated)
+	if r.Error != nil {
+		klog.Errorf("error querying last update time: %s", r.Error)
+		return
+	}
+	klog.Infof("ran the last update query: %s", lastUpdated)
+
+	// Load all the job reports for this release to calculate statistics:
+	filterOpts := &filter.FilterOptions{
+		Filter:    &filter.Filter{},
+		SortField: "current_pass_percentage",
+		Sort:      apitype.SortDescending,
+		Limit:     0,
+	}
+	start := time.Now().Add(-14 * 24 * time.Hour)
+	boundary := time.Now().Add(-7 * 24 * time.Hour)
+	end := time.Now()
+	jobReports, err := query.JobReports(dbc, filterOpts, release, start, boundary, end)
+	if err != nil {
+		klog.Errorf("error querying job reports: %s", err)
+		return
+	}
+	currStats, prevStats := calculateJobResultStatistics(jobReports)
+
+	RespondWithJSON(http.StatusOK, w, health{
+		Indicators:  indicators,
+		LastUpdated: lastUpdated,
+		Current:     currStats,
+		Previous:    prevStats,
+		/*
+				// NOTE: looks unused in ReleaseOverview and of questionable use in general. This field would show
+				// the number of variants whose jobs fully pass over 80% of the time, 60% of the time, or less than that.
+				// skipping for now.
+					Variants: variants{
+						Current:  curr.TopLevelIndicators.Variant,
+						Previous: prev.TopLevelIndicators.Variant,
+					},
+
+			    // TODO: do we have need of these anymore with the new installer tests coming from junit?
+					Warnings: append(curr.AnalysisWarnings, prev.AnalysisWarnings...),
```

Comment:
That's just one warning, but there are some others. For example we show one when the last 3 promotion jobs failed.  We need to figure out how to generate and display warnings from the DB-side of things, but I guess it can be done separately. Additional warnings will need to include things like OpenShift/RHCOS release mismatch (there's a card for that one)